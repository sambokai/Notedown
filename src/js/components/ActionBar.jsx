/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import PropTypes from 'prop-types';


// eslint-disable-next-line react/prefer-stateless-function
class ActionBar extends React.Component {
  render() {
    return (
      <nav className="navbar navbar-dark bg-dark">
        <a className="navbar-brand" href="#">Notedown</a>
        <button
          onClick={this.props.onCreateNote}
          type="button"
          className="btn btn-primary"
        >New Note
        </button>
      </nav>
    );
  }
}

ActionBar.propTypes = {
  onCreateNote: PropTypes.func.isRequired,
};


export default ActionBar;
