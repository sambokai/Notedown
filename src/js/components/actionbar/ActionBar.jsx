/* eslint-disable jsx-a11y/anchor-is-valid */
import React from 'react';
import PropTypes from 'prop-types';


// eslint-disable-next-line react/prefer-stateless-function
class ActionBar extends React.Component {
  render() {
    return (
      <nav className="navbar navbar-dark bg-dark justify-content-between">
        <a id="navbar-logo" className="navbar-brand" href="/">Notedown</a>
        <form className="form-inline">
          <button
            id="new-note-button"
            onClick={this.props.onCreateNote}
            type="button"
            className={`btn btn-primary ml-2 ${this.props.noteCreationAllowed ? '' : 'disabled'}`}
          >New Note
          </button>
          <button
            id="delete-note-button"
            onClick={this.props.onDeleteNote}
            type="button"
            className="btn btn-danger ml-2"
          >Delete
          </button>
        </form>
      </nav>
    );
  }
}

ActionBar.propTypes = {
  onCreateNote: PropTypes.func.isRequired,
  onDeleteNote: PropTypes.func.isRequired,
  noteCreationAllowed: PropTypes.bool.isRequired,
};


export default ActionBar;
