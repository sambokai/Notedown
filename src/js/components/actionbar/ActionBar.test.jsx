import React from 'react';

import renderer from 'react-test-renderer';
import { shallow } from 'enzyme/build/index';

import ActionBar from './ActionBar';

const onCreateNoteMock = jest.fn();
const onDeleteNoteMock = jest.fn();

const mockPropCallbacks = {
  onCreateNote: onCreateNoteMock,
  onDeleteNote: onDeleteNoteMock,
};

describe('<ActionBar/>', () => {
  it('renders correctly with delete-button enabled if creation of new notes is allowed', () => {
    const wrapper = renderer
      .create(<ActionBar {...mockPropCallbacks} noteCreationAllowed />);

    expect(wrapper)
      .toMatchSnapshot();
  });

  it('renders correctly with delete-button disabled if creation of new notes is forbidden', () => {
    const wrapper = renderer
      .create(<ActionBar {...mockPropCallbacks} noteCreationAllowed={false} />);

    expect(wrapper)
      .toMatchSnapshot();
  });

  it("calls onCreateNote callback, upon click on the 'New Note' button ", () => {
    const wrapper = shallow(<ActionBar {...mockPropCallbacks} noteCreationAllowed />);

    wrapper.find('#new-note-button').simulate('click');
    expect(mockPropCallbacks.onCreateNote.mock.calls).toHaveLength(1);
  });

  it("calls onDeleteNote callback, upon click on the 'Delete' button ", () => {
    const wrapper = shallow(<ActionBar {...mockPropCallbacks} noteCreationAllowed />);

    wrapper.find('#delete-note-button').simulate('click');
    expect(mockPropCallbacks.onDeleteNote.mock.calls).toHaveLength(1);
  });
});
