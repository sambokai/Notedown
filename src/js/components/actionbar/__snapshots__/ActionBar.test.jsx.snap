// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ActionBar/> renders correctly with delete-button disabled if creation of new notes is forbidden 1`] = `
<nav
  className="navbar navbar-dark bg-dark justify-content-between"
>
  <a
    className="navbar-brand"
    href="/"
    id="navbar-logo"
  >
    Notedown
  </a>
  <form
    className="form-inline"
  >
    <button
      className="btn btn-primary ml-2 disabled"
      id="new-note-button"
      onClick={[MockFunction]}
      type="button"
    >
      New Note
    </button>
    <button
      className="btn btn-danger ml-2"
      id="delete-note-button"
      onClick={[MockFunction]}
      type="button"
    >
      Delete
    </button>
  </form>
</nav>
`;

exports[`<ActionBar/> renders correctly with delete-button enabled if creation of new notes is allowed 1`] = `
<nav
  className="navbar navbar-dark bg-dark justify-content-between"
>
  <a
    className="navbar-brand"
    href="/"
    id="navbar-logo"
  >
    Notedown
  </a>
  <form
    className="form-inline"
  >
    <button
      className="btn btn-primary ml-2 "
      id="new-note-button"
      onClick={[MockFunction]}
      type="button"
    >
      New Note
    </button>
    <button
      className="btn btn-danger ml-2"
      id="delete-note-button"
      onClick={[MockFunction]}
      type="button"
    >
      Delete
    </button>
  </form>
</nav>
`;
